src/io/flutter/run/daemon/DaemonApi.java

```diff
@@ -236,6 +203,64 @@ private Command takePending(int id) {
     }
   }
 
+  /**
+   * Parse the given string; if it's valid JSON, and a valid Daemon response of event, then return
+   * the parsed JsonObject.
+   */
+  static JsonObject parseAndValidateDaemonEvent(String str) {
```

Comment:
Maybe renaming `str` to `event` would help here?

Reply:
Or `eventString`...

Reply:
==> `message`